stdlib/log-api/src/main/ballerina/log/natives.bal

```diff
@@ -18,25 +18,35 @@
 # Logs the specified value at DEBUG level.
 #
 # + msg - The message to be logged
-public extern function printDebug(string msg);
+# + func - lambda function to be executed
+# + args - Arguments to be passed for the placeholders of the log message
+public extern function printDebug(string msg, (function() returns (string))? func = (), any... args);
```

Comment:
I think correct signaure shouled be,  `function printDebug(string|(function() returns (string)) message);` ether you provide a string value. or provide a function pointer, which returns the message.

Ref args: I think we don't need to add this, because users can use sprintf function to do the same thing.

Reply:
Ops.. wasn't aware of `io:sprintf` function. +1 will update the PR.